### ISSUE TITLE
Piximap program crash

### ISSUE THREAD

**Issue by JordanGarske** [pymupdf/PyMuPDF#3848]
### Description of the bug

What is happening is that when I read from the PDF, I use the rectangle information to collect color data. Recently, however, I encountered an issue with one of the PDF collections: while collecting the color information, the program crashes unexpectedly without any error messages. I traced the problem to the color_topusage function. Specifically, the error appears to originate from the call to this function, where the issue seems to be related to color_count -> JM_color_count -> oldpix = read_sample(pm, s, n).

### How to reproduce the bug

 I traced the problem to the color_topusage function. Specifically, the error appears to originate from the call to this function, where the issue seems to be related to color_count -> JM_color_count -> oldpix = read_sample(pm, s, n).

### PyMuPDF version

1.24.10

### Operating system

Windows

### Python version

3.12

**Comment by JorjMcKie** [comment]
Please do provide reproducing file and code.

**Comment by JordanGarske** [comment]
[case.pdf](https://github.com/user-attachments/files/16894188/case.pdf)
import pymupdf
from pymupdf import  Page, table, Rect , Pixmap, sRGB_to_rgb

from pymupdf.utils import get_pixmap, get_page_pixmap
#Similar file with error; it loops on a 4-page document and crashes 

on page 2 of 4.
pdf = pymupdf.open(f'./case.pdf')
for i in range(0, len(pdf) ):
    page = pdf.load_page(i)
    print(page)
    for annot in page.get_drawings():
            if page.get_textbox(annot['rect'] ) :
                rect = annot['rect']
                pix: Pixmap = get_pixmap(page, clip= rect)
                colorBytes = pix.color_topusage()
print('done')

**Comment by JorjMcKie** [comment]
This script works:
```python
import pymupdf

doc = pymupdf.open("case.pdf")
page = doc[1]
paths = page.get_drawings()
for i, p in enumerate(paths):
    rect = p["rect"]  # best check for emptiness already here!
    text = page.get_textbox(rect)
    if text:
        pix = page.get_pixmap(clip=rect)
        if pymupdf.IRect(pix.irect).is_empty:
            print(i, "pixmap has empty area - skipping")
            continue
        print("path", i, pix.color_topusage())
```
So the reason for the crash is a pixmap that covers no area, and the `color_topusage()` method does not protext itself against nonsense input.
Which must be fixed of course: we will raise an exception in such cases.

Otherwise please take note of the following comments:
* You are iterating over vector graphics which can consist of just a line - which obviously cannot contain anything, including text. You could / should ignore those right away. Lines may be not parallel to either the x- or the y-axis. In that case, the `p["rect"]`is not empty, so everything appears to work ... but still the underlying line contains no text, while p["rect"] **_may_** have text inside. Can't imagine you want those false alarms.
* Depending on the circumstances (please read the relevant documentation about Rect / IRect objects and their relationship) a non-empty rectangle may be wrapped by an empty IRect (!).
* Your use of the term "annotation" may lead to misconceptions: this is a PDF term reserved for something entirely different. Best stay with line-art, vector graphic or drawing.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.13.
